src/Cache/CacheOperationTypes/CacheOperationTypes.php

```diff
@@ -982,47 +982,46 @@ public function asError(): CacheDictionaryGetFieldsResponseError|null
 
 class CacheDictionaryGetFieldsResponseSuccess extends CacheDictionaryGetFieldsResponse
 {
-    private array $responsesList = [];
+    private array $responses = [];
+    private array $valuesDictionary = [];
 
-    public function __construct(_DictionaryGetResponse $responses = null, ?int $numRequested = null)
+    public function __construct(_DictionaryGetResponse $responses = null, ?int $numRequested = null, ?array $fields = null)
     {
         if (!is_null($responses) && is_null($numRequested)) {
+            $counter = 0;
             parent::__construct();
             foreach ($responses->getFound()->getItems() as $response) {
                 if ($response->getResult() == ECacheResult::Hit) {
-                    $this->responsesList[] = new CacheDictionaryGetFieldResponseHit(null, $response->getCacheBody());
-                }
-                if ($response->getResult() == ECacheResult::Miss) {
-                    $this->responsesList[] = new CacheDictionaryGetFieldResponseMiss();
+                    $this->responses[] = new CacheDictionaryGetFieldResponseHit(null, $response->getCacheBody());
+                    $this->valuesDictionary[$fields[$counter]] = $response->getCacheBody();
+                    $counter++;
+                } else if ($response->getResult() == ECacheResult::Miss) {
+                    $this->responses[] = new CacheDictionaryGetFieldResponseMiss();
                 } else {
-                    $this->responsesList[] = new CacheDictionaryGetFieldResponseError(new UnknownError(strval($response->getResult())));
+                    $this->responses[] = new CacheDictionaryGetFieldResponseError(new UnknownError(strval($response->getResult())));
                 }
             }
         }
         if (is_null($responses) && !is_null($numRequested)) {
             foreach (range(0, $numRequested - 1) as $ignored) {
-                $this->responsesList[] = new CacheDictionaryGetFieldResponseMiss();
+                $this->responses[] = new CacheDictionaryGetFieldResponseMiss();
```

Comment:
I think all of this is the result of trying to use a `Success` class where we need to be using `Hit` and `Miss` classes instead. I left [a comment](https://github.com/momentohq/client-sdk-php/pull/65/files#r1024591717) on `_ScsDataClient` that lines up with this one.

Reply:
When I moved `CacheDicitonaryGetBatch` from our .NET incubating repo, I referenced [here](https://github.com/momentohq/client-sdk-dotnet-incubating/blob/main/src/Momento.Sdk.Incubating/Responses/CacheDictionaryGetBatchResponse.cs), which has `Success` and `Error` as `CacheDictionaryGetBatchResponse`.
Sorry if I missed, but are we going to update this response type to `Hit`, `Miss`, and `Error` instead?

Reply:
You might double check with @cprice404 but I think this should follow the Hit/Miss/Error pattern since we're dealing with a single dictionary value and we can tell if our request was a miss in the corresponding data client code. It should also allow us to get rid of the `$numRequested` parameter and logic in the `CacheDictionaryGetFieldsResponseSuccess` constructor.

Reply:
I think I agree with @pgautier404 , I think this was an oversight when we were switching the .NET SDK over.

Here:

https://github.com/momentohq/client-sdk-dotnet-incubating/blame/main/src/Momento.Sdk.Incubating/Internal/ScsDataClient.cs#L312

That code does not check for the "Missing" case that we got from the server.  My gut is that it should be checking for that case, and should use Hit/Miss/Error pattern instead of success.  That's what we do for DictionaryFetch, ListFech, etc.  So we should fix that both in .NET and PHP.  I created this ticket to capture it for .NET https://github.com/momentohq/client-sdk-dotnet-incubating/issues/47 but yes let's go ahead and make this change here.  @malandis if you know of any reason why DictionaryGetBatch/DictionaryGetFields shouldn't use Hit/Miss/Error pattern let us know.

I didn't review this closely enough to understand why that change would eliminate the need for the numRequested/counter thing but I also agree that it seems like we shouldn't need those, so if they still seem necessary after the switch to Hit/Miss/Error feel free to ping for 👀 .

Sorry for not catching this during the .NET work.

Reply:
@pgautier404 @cprice404 @malandis 
My understanding is we want to have 4 different response patters:

1. We found a dictionary and all items in the dictionary also found.
2. We found a dictionary but some of the items in the dictionary are missing.
3. We did not find a dictionary.
4. We had an error finding a dictionary.

Cases 1, 3, 4 are straightforward.
1 -> Hit
3 -> Miss
4 -> Error

But how about case 2? Would this be Hit although some of the items are Miss?
So Hit means that finding a dictionary not necessarily the items inside?

Reply:
yes my opinion is that if the Dictionary itself is found, then case 2 is a HIT.

users may then check for individual fields by iterating over the `responses` and seeing which are HIT vs MISS, or they can just use `valueDictionary` and check to see if each key exists.